projects/igniteui-angular/src/lib/directives/for-of/for_of.directive.ts

```diff
@@ -129,6 +129,23 @@ export class IgxForOfDirective<T> implements OnInit, OnChanges, DoCheck, OnDestr
      */
     public totalItemCount: number = null;
 
+    /**
+     * A function that returns whether the element is the first or not
+     */
+    public first = (index) => { return index === 0};
+    /**
+     * A function that returns whether the element is the last or not
+     */
+    public last = (index) => { return index === this.igxForOf.length};
```

Comment:
arrays are 0-based so the last index is length - 1

Reply:
Done